src/audio/dcblock/dcblock.c

```diff
@@ -0,0 +1,434 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2020 Google LLC. All rights reserved.
+//
+// Author: Sebastiano Carlucci <scarlucci@google.com>
+
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/audio/format.h>
+#include <sof/audio/pipeline.h>
+#include <sof/audio/dcblock/dcblock.h>
+#include <sof/common.h>
+#include <sof/debug/panic.h>
+#include <sof/drivers/ipc.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/memory.h>
+#include <sof/list.h>
+#include <sof/platform.h>
+#include <sof/string.h>
+#include <sof/trace/trace.h>
+#include <ipc/control.h>
+#include <ipc/stream.h>
+#include <ipc/topology.h>
+#include <user/trace.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/**
+ * \brief Sets the DC Blocking filter in pass through mode.
+ * The frequency response of a DCB filter is:
+ * H(z) = (1 - z^-1)/(1-Rz^-1).
+ * Setting R to 1 makes the filter act as a passthrough component.
+ */
+static int dcblock_set_passthrough(struct comp_data *cd)
+{
+	comp_cl_info(&comp_dcblock, "dcblock_set_passthrough()");
+	int i;
+
+	for (i = 0; i < PLATFORM_MAX_CHANNELS; i++)
+		cd->R_coeffs[i] = ONE_Q2_30;
+
+	return 0;
+}
+
+/**
+ * \brief Initializes the state of the DC Blocking Filter
+ */
+static int dcblock_init_state(struct comp_data *cd,
+			      struct audio_stream *src_stream)
+{
+	int32_t *x0;
+	int i;
+
+	for (i = 0; i < PLATFORM_MAX_CHANNELS; i++) {
+		x0 = audio_stream_read_frag_s32(src_stream, 0);
+		cd->state[i].y_prev = 0;
+
+		switch (src_stream->frame_fmt) {
+		case SOF_IPC_FRAME_S16_LE:
```

Comment:
Checking the frame format on every loop iteration is not efficient, best to check it once at the start.

Reply:
@lgirdwood Updated